File: arrays/squares_of_a_sorted_array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def sorted_squares(nums)
    left = 0
    right = nums.size - 1
    result = []

    while left <= right
        mid = (left  + right) / 2
        if nums[left].abs > nums[right].abs
            result << nums[left] ** 2
            left += 1
        else
            result << nums[right] ** 2
            right -= 1
        end
    end

    result.reverse
end

File: arrays/squares_of_a_sorted_array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        result = []

        while left <= right:
            mid = (left + right) // 2
            if abs(nums[left]) > abs(nums[right]):
                result.append(nums[left] ** 2)
                left += 1
            else:
                result.append(nums[right] ** 2)
                right -= 1

        result.reverse()
        return result